frontend/src/components/common/Resource/ResourceTable.tsx

```diff
@@ -39,16 +40,57 @@ export default function ResourceTable(
   return <Table {...(props as ResourceTableProps)} />;
 }
 
+/**
+ * Returns a throttled version of the input value.
+ *
+ * @param value - The value to be throttled.
+ * @param interval - The interval in milliseconds to throttle the value.
+ * @returns The throttled value.
+ */
+export function useThrottle(value: any, interval = 1000): any {
+  const [throttledValue, setThrottledValue] = useState(value);
+  const lastEffected = useRef(Date.now() + interval);
+
+  // Ensure we don't throttle holding the loading null or undefined value before
+  // real data comes in. Otherwise we could wait up to interval milliseconds
+  // before we update the throttled value.
+  //
+  //   numEffected == 0,  null, or undefined whilst loading.
+  //   numEffected == 1,  real data.
+  const numEffected = useRef(0);
+
+  useEffect(() => {
+    const now = Date.now();
+
+    if (now >= lastEffected.current + interval || numEffected.current < 2) {
+      numEffected.current = numEffected.current + 1;
```

Comment:
Would it be more readable if `numEffected` was just `doThrottle = useRef(false)` and we'd set it to true if `!!value`.
So it be `if (now >= lastEffected.current + interval || !doThrottle.current)`.

Would this be logically equivalent?

Reply:
🤷 The way it's done makes it more apparent it's a counter.

Reply:
It's a difficult to understand 1-2 counter that checks for whether we've finally have content, but in the interest of time I will merge it.